lib/ui/employee_id/employee_id_card.dart

```diff
@@ -41,7 +30,10 @@ class _EmployeeIdCardState extends State<EmployeeIdCard> {
       isLoading: Provider.of<EmployeeIdDataProvider>(context).isLoading,
       titleText: CardTitleConstants.titleMap[cardId],
       errorText: Provider.of<EmployeeIdDataProvider>(context).error,
-      child: () => isValidId
+      child: () => (employeeModel != null &&
```

Comment:
Consider putting this into a static private method unless it is only used in this function just once.